dask/dataframe/io/tests/test_orc.py

```diff
@@ -75,8 +79,110 @@ def test_orc_single(orc_files):
 
 
 def test_orc_multiple(orc_files):
-    d = read_orc(orc_files[0])
-    d2 = read_orc(orc_files)
+    d = dd.read_orc(orc_files[0])
+    d2 = dd.read_orc(orc_files)
     assert_eq(d2[columns], dd.concat([d, d])[columns], check_index=False)
-    d2 = read_orc(os.path.dirname(orc_files[0]) + "/*.orc")
+    d2 = dd.read_orc(os.path.dirname(orc_files[0]) + "/*.orc")
     assert_eq(d2[columns], dd.concat([d, d])[columns], check_index=False)
+
+
+@pytest.mark.skipif(
+    parse_version(pa.__version__) < parse_version("4.0.0"),
+    reason=("PyArrow>=4.0.0 required for ORC write support."),
+)
+@pytest.mark.parametrize("index", [None, "i32"])
+@pytest.mark.parametrize("columns", [None, ["i32", "i64", "f"]])
+def test_orc_roundtrip(tmpdir, index, columns):
+    tmp = str(tmpdir)
+    data = pd.DataFrame(
+        {
+            "i32": np.arange(1000, dtype=np.int32),
+            "i64": np.arange(1000, dtype=np.int64),
+            "f": np.arange(1000, dtype=np.float64),
+            "bhello": np.random.choice(["hello", "yo", "people"], size=1000).astype(
+                "O"
+            ),
+        }
+    )
+    if index:
+        data.set_index(index, inplace=True)
+    df = dd.from_pandas(data, chunksize=500)
+    if columns:
+        data = data[[c for c in columns if c != index]]
+
+    # Write
+    df.to_orc(tmp, write_index=bool(index))
+
+    # Read
+    df2 = dd.read_orc(tmp, index=index, columns=columns)
+    assert_eq(data, df2, check_index=bool(index))
+
+
+@pytest.mark.skipif(
+    parse_version(pa.__version__) < parse_version("4.0.0"),
+    reason=("PyArrow>=4.0.0 required for ORC write support."),
+)
+@pytest.mark.parametrize("split_stripes", [True, False, 2, 4])
+def test_orc_roundtrip_aggregate_files(tmpdir, split_stripes):
+    tmp = str(tmpdir)
+    data = pd.DataFrame(
+        {
+            "a": np.arange(100, dtype=np.float64),
+            "b": np.random.choice(["cat", "dog", "mouse"], size=100),
+        }
+    )
+    df = dd.from_pandas(data, npartitions=8)
+    df.to_orc(tmp, write_index=False)
+    df2 = dd.read_orc(tmp, split_stripes=split_stripes, aggregate_files=True)
+
+    # Check that we have the expected partition count
+    # and that the data is correct
+    if split_stripes:
+        assert df2.npartitions == df.npartitions / int(split_stripes)
+    else:
+        assert df2.npartitions == df.npartitions
+    assert_eq(data, df2, check_index=False)
+
+
+@pytest.mark.skipif(
+    sys.version_info[:2] > (3, 8),
+    reason=("PyORC sometimes crashes for python-3.9"),
+)
```

Comment:
Second try...  It looks like pyorc has some [libprotobuf-collision problems](https://github.com/noirello/pyorc/issues/10) that seem to crash the worker for python-3.9 on macos.  For now, I am just skipping the problematic test for python-3.9.  Hopefully this is reasonable.